sdk/core/Azure.Core/src/Internal/InitializationConstructorAttribute.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// An attribute class indicating to AutoRest which constructor to use for initialization.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Constructor)]
+    internal class InitializationConstructorAttribute : Attribute
```

Comment:
Don't these types already exist in the repo for the mgmt plane usage? Should we just move those into core and pull them into mgmt projects as needed?

Reply:
We thought about making these internal shared classes, but wanted to move that out of scope for this change since we don't necessarily want to add to the existing circular internal shared reference issue.

Reply:
I could do the first version of breaking this dependency if we feel its needed for this, but IMHO this can be done after GA since its all internal code

Reply:
Can you explain what you mean about the existing circular reference issue?

Reply:
instead of an SDK pulling shared source from azure.core (its dependency), Azure.Core gives its files to Autorest, the SDK references Azure.Core but pulls all of the shared Azure.Core files from Autorest.